src/main/cpp/hierarchy.cpp

```diff
@@ -433,6 +433,22 @@ void Hierarchy::updateChildren(ProvisionNode& pn, const LoggerPtr& logger)
 			l->setParent( logger );
 		}
 	}
+    
+}
+
+void Hierarchy::updateChildren(const Logger* parent)
+{
+	for (auto& item : m_priv->loggers)
+	{
+		for (auto l = item.second; l; l = l->getParent())
+		{
+			if (l->getParent().get() == parent)
```

Comment:
I'm pretty sure this is going to only go one level deep though?  so `com.example` would get set but not `com.example.foo`

Reply:
Oh yes you're right, I misread what the code was doing.